sphinx_gallery/py_source_parser.py

```diff
@@ -19,30 +19,57 @@
 """
 
 
-def get_docstring_and_rest(filename):
-    """Separate `filename` content between docstring and the rest
+def parse_file_source(filename):
+    """Parse source file into AST node
+
+    Parameters
+    ----------
+    filename : str
+        File path
 
-    Strongly inspired from ast.get_docstring.
 
     Returns
     -------
-    docstring: str
-        docstring of `filename`
-    rest: str
-        `filename` content without the docstring
+    node : AST node
+    content : utf-8 encoded string
     """
+
     # can't use codecs.open(filename, 'r', 'utf-8') here b/c ast doesn't
-    # seem to work with unicode strings in Python2.7
-    # "SyntaxError: encoding declaration in Unicode string"
+    # work with unicode strings in Python2.7 "SyntaxError: encoding
+    # declaration in Unicode string" In python 2.7 the string can't be
+    # encoded and have information about its encoding. That is particularly
+    # problematic since source files include in their header information
+    # about the file encoding.
+    # Minimal example to fail: ast.parse(u'# -*- coding: utf-8 -*-')
+
     with open(filename, 'rb') as fid:
         content = fid.read()
     # change from Windows format to UNIX for uniformity
     content = content.replace(b'\r\n', b'\n')
 
     try:
         node = ast.parse(content)
+        return node, content.decode('utf-8')
     except SyntaxError:
-        return SYNTAX_ERROR_DOCSTRING, content.decode('utf-8'), 1
+        return SyntaxError, content.decode('utf-8')
+
+
+def get_docstring_and_rest(filename):
+    """Separate `filename` content between docstring and the rest
```

Comment:
I think you need to use double backticks if you want this to be fixed-width in the HTML reference documentation. Same thing for a few single backticks that you use in a few different places.